src/scanner.cc

```diff
@@ -48,6 +48,10 @@ struct Scanner {
     return tokens_.back();
   }
 
+  bool atTopLevel() {
+    return peek() == -1;
```

Comment:
I expected the stack to be empty at top level and I was surprised to see that it actually contains a `-1` element. It looks like it's pushed by the `deserialize()` method. Can we count on the state of the stack to be consistently initialised by that method or should we also plan for an empty stack?

Reply:
Having the constructor initialize with `-1` seems reasonable. We effectively do that right now by just calling `deserialize` on the "empty" object created here.

https://github.com/r-lib/tree-sitter-r/blob/6a43a147153ee5be692b1390c2dc01f1b8397606/src/scanner.cc#L279-L283

The intention is that `pop()` always takes a token type, and it should never accept the value `-1` (although we don't have a way of enforcing that). In other words, if you try to 'pop' a token type that doesn't exist on the stack, then the `pop` doesn't happen. (The method should probably be called `maybePop()` or `popIfMatching()`.)